src/reducers/status/reducers.js

```diff
@@ -0,0 +1,48 @@
+import { calculateWinner } from '../../components/game/controller';
+
+import { SELECT_SQUARE, GO_TO_MOVE } from './actions';
+
+export const initialState = {
+  history: [
+    {
+      squares: Array(9).fill(''),
+    },
+  ],
+  xIsNext: true,
+  moveNumber: 0,
+};
+
+export default (state = initialState, action = {}) => {
+  const history = state.history.slice(0, state.moveNumber + 1);
+  const current = history[state.moveNumber];
+  const squares = current.squares.slice();
+  switch (action.type) {
+    case GO_TO_MOVE:
+      return {
+        ...state,
+        history: state.history.slice(0, action.step + 1),
+        moveNumber: action.step,
+        xIsNext: !(action.step % 2),
+      };
+
+    case SELECT_SQUARE:
+      if (calculateWinner(squares) || squares[action.index]) {
+        return state;
+      }
```

Comment:
You can move this functionality further up. You don't need to dispatch an action if the `if` statement criteria is met.